types/yargs/index.d.mts

```diff
@@ -8,7 +8,11 @@ interface MainType {
   filename: string;
   children: MainType[];
 }
-declare const _instanceFactory: (processArgs: string[], cwd?: string, parentRequire?: RequireType) => yargs.Argv;
+declare const _instanceFactory: (
+  processArgs?: ReadonlyArray<string> | string,
```

Comment:
Currently, ts returns an error if a string or no value is passed to the yargs constructor; however, doing so is valid when using yargs.

[yargs source code](https://github.com/yargs/yargs/blob/8912078ae5e2157f5b78298a2ceb5cdf41518e85/lib/yargs-factory.ts#L67)

## Examples

All three of these are valid ways to execute yargs

```ts
import yargs from 'yargs'

// yargs with string

yargs('add 1 2')
  .command(
    "add <a> <b>",
    "add desc",
    (yargs) =>
      yargs
        .positional("a", {
          desc: "a desc",
          type: "number",
        })
        .positional("b", {
          desc: "a desc",
          type: "number",
        }),
    (argv) => {
      const { a, b } = argv as { a: number; b: number };
      console.log(`${a} + ${b} = ${a + b}`);
    },
  ).parse()

// yargs with array

yargs(['add', '1', '2'])
  .command(
    "add <a> <b>",
    "add desc",
    (yargs) =>
      yargs
        .positional("a", {
          desc: "a desc",
          type: "number",
        })
        .positional("b", {
          desc: "a desc",
          type: "number",
        }),
    (argv) => {
      const { a, b } = argv as { a: number; b: number };
      console.log(`${a} + ${b} = ${a + b}`);
    },
  ).parse()

// parse with string (also works with array)

yargs()
  .command(
    "add <a> <b>",
    "add desc",
    (yargs) =>
      yargs
        .positional("a", {
          desc: "a desc",
          type: "number",
        })
        .positional("b", {
          desc: "a desc",
          type: "number",
        }),
    (argv) => {
      const { a, b } = argv as { a: number; b: number };
      console.log(`${a} + ${b} = ${a + b}`);
    },
  ).parse('add 1 2')
```